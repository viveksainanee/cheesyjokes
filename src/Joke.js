import React, { Component } from 'react';
import { library } from '@fortawesome/fontawesome-svg-core';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faThumbsUp,
  faThumbsDown,
  faStroopwafel
} from '@fortawesome/free-solid-svg-icons';

library.add(faThumbsUp, faThumbsDown, faStroopwafel);

class Joke extends Component {
  constructor(props) {
    super(props);
    this.handleIncrement = this.handleIncrement.bind(this);
    this.handleDecrement = this.handleDecrement.bind(this);
  }

  handleIncrement() {
    this.props.increment(this.props.id);
  }

  handleDecrement() {
    this.props.decrement(this.props.id);
  }

  render() {
    return (
      <div key={this.props.id} className="Joke">
        <div>{this.props.text}</div>

        <button onClick={this.handleIncrement}>
          <FontAwesomeIcon icon="thumbs-up" />
        </button>

        {this.props.up}
        <button onClick={this.handleDecrement}>
          <FontAwesomeIcon icon="thumbs-down" />
        </button>
        {this.props.down}

        <div> Net Score: {+this.props.up + +this.props.down}</div>
      </div>
    );
  }
}

export default Joke;
