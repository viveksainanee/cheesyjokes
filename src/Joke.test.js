import React from 'react';
import { shallow, mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import Joke from './Joke';

it('renders shallowly without crashing', function() {
  shallow(<Joke text="Knock knock" id="2" up={0} down={0} />);
});

it('passes snapshot test', function() {
  let wrapper = shallow(<Joke text="Knock knock" id="2" up={0} down={0} />);
  let serialized = toJSON(wrapper);
  expect(serialized).toMatchSnapshot();
});
