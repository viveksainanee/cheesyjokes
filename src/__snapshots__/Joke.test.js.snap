// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`passes snapshot test 1`] = `
<div
  className="Joke"
  key="2"
>
  <div>
    Knock knock
  </div>
  <button
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="thumbs-up"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
  0
  <button
    onClick={[Function]}
  >
    <FontAwesomeIcon
      border={false}
      className=""
      fixedWidth={false}
      flip={null}
      icon="thumbs-down"
      inverse={false}
      listItem={false}
      mask={null}
      pull={null}
      pulse={false}
      rotation={null}
      size={null}
      spin={false}
      symbol={false}
      title=""
      transform={null}
    />
  </button>
  0
  <div>
     Net Score: 
    0
  </div>
</div>
`;
